File: Max Sum Subarray of size K - GFG/max-sum-subarray-of-size-k.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends

#define ll long long;
class Solution{  
public:
    long maximumSumSubarray(int k, vector<int>&arr , int N){
        long  sum=0;
        int i=0;
        int j=0;
        for(;i<k;i++){
            sum+=arr[i];
        }
        long maxi=sum;
        for(i=k;i<N;i++){
            sum-=arr[j];
            j++;
            sum+=arr[i];
            maxi=max(sum,maxi);
        }
        return maxi;
    }
};

//{ Driver Code Starts.
int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int N,K;
        cin >> N >> K;;
        vector<int>Arr;
        for(int i=0;i<N;++i){
            int x;
            cin>>x;
            Arr.push_back(x);
        }
        Solution ob;
        cout << ob.maximumSumSubarray(K,Arr,N) << endl;
    }
    return 0; 
} 
// } Driver Code Ends

File: Max Sum Subarray of size K - GFG/max-sum-subarray-of-size-k.java
//{ Driver Code Starts
//Initial Template for Java
import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-- > 0)
        {
            
            int N=sc.nextInt();
            int K=sc.nextInt();
            ArrayList<Integer> Arr = new ArrayList<Integer>(N); 
  
            // Appending new elements at 
            // the end of the list 
            for (int i = 0; i < N; i++) {
                int x = sc.nextInt();
                Arr.add(x); 
            }

            Solution ob = new Solution();
            System.out.println(ob.maximumSumSubarray(K,Arr,N));
        }
    }
}

// } Driver Code Ends


//User function Template for Java
class Solution{
    static long maximumSumSubarray(int k, ArrayList<Integer> arr,int N){
        long sum=0;
        int i=0;
        int j=0;
        for(;i<k;i++){
            sum+=arr.get(i);
        }
        long max=sum;
        for(i=k;i<N;i++){
            sum-=arr.get(j);
            j++;
            sum+=arr.get(i);
            max=Math.max(sum,max);
        }
        return max;
    }
}